src/PrettyPrinting.jl

```diff
@@ -1591,23 +1591,89 @@ function _write_line(io::IOCustom, str::AbstractString)
     spaceleft = c - ind - io.printed
   end
   #@show spaceleft
-  firstlen = min(spaceleft, length(str))
-  firststr = str[1:firstlen]
-  if io.lowercasefirst
-    written += write(io.io, lowercasefirst(firststr))
-    io.lowercasefirst = false
+  _graphemes = Base.Unicode.graphemes(str)
+  firstlen = min(spaceleft, length(_graphemes))
+  # make an iterator over valid indices
+  firstiter = Base.Iterators.take(_graphemes, firstlen)
+  restiter = Base.Iterators.drop(_graphemes, firstlen)
```

Comment:
is this still necessary? isn't `_graphemes` a simple `Vector`? Then perhaps this also works:
```suggestion
  firstiter = _graphemes[1:firstlen]
  restiter = _graphemes[firstlen+1:end]
```

And then you could rename `firstiter` and `restiter` and further simplify some code using them (e.g. `collect` on them now would be redundant)

Reply:
No, unfortunately `_graphemes` is an iterator.

`graphemes(s::AbstractString) -> GraphemeIterator` ( https://docs.julialang.org/en/v1/stdlib/Unicode/#Unicode.graphemes )